fix(ai): Actually use todo examples defined on mode class

Twixes opened the pull request:
## Problem

The examples for mode-specific planning with the `todo_write` tool weren't actually being used when set as `POSITIVE_TODO_EXAMPLES`/`NEGATIVE_TODO_EXAMPLES` on the mode class! Instead of `cls.POSITIVE_TODO_EXAMPLES`, the global var `POSITIVE_TODO_EXAMPLES` was used – so only calling `create_tool_class()` would have worked, but this wasn't in use. I checked traces.

## Changes

Fixed `TodoItem`.

Twixes committed:
fix(ai): Actually use todo examples defined on mode class

skoob13 reviewed (approved):
Thanks!

skoob13 reviewed (changes requested):
Seems like it doesn't work either

skoob13 committed:
fix: todo_write example accumulation

skoob13 reviewed (approved):
Pushed a commit that should fix it!

Twixes committed:
Fix example formatting

Looks like it should be in this format: https://github.com/PostHog/posthog/blob/fb3caba32371ac50d2ddebf1d6d43bb51951091e/products/experiments/backend/max_tools.py#L174-L185

Twixes committed:
Make mypy happy
